core-common/src/test/java/org/glassfish/jersey/internal/ExceptionMapperFactoryTest.java

```diff
@@ -0,0 +1,550 @@
+     */
+    private static class IllegalArgumentExceptionMapper implements ExtendedExceptionMapper<RuntimeException> {
+
+        @Override
+        public boolean isMappable(RuntimeException exception) {
+            return exception instanceof IllegalArgumentException;
+        }
+
+        @Override
+        public Response toResponse(RuntimeException exception) {
+            return Response
+                    .status(Response.Status.BAD_REQUEST)
+                    .build();
+        }
+
+    }
+
+    /**
+     * Extended Exception Mapper which has RuntimeException as generic type and isMappable returns true if the
+     * exception is instance of IllegalStateException.
+     */
+    private static class IllegalStateExceptionMapper implements ExtendedExceptionMapper<RuntimeException> {
+
+        @Override
+        public boolean isMappable(RuntimeException exception) {
+            return exception instanceof IllegalStateException;
+        }
+
+        @Override
+        public Response toResponse(RuntimeException exception) {
+            return Response
+                    .status(Response.Status.SERVICE_UNAVAILABLE)
+                    .build();
+        }
+
+    }
+
+    /**
+     * Exception Mapper which has RuntimeException as generic type.
+     */
+    private static class RuntimeExceptionMapper implements ExceptionMapper<RuntimeException> {
+
+        @Override
+        public Response toResponse(RuntimeException exception) {
+            return Response
+                    .status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .build();
+        }
+
+    }
+
+
+    /**
+     * Stub implementation -- reason for this is to control the order of list of services in ServiceLocator.
+     *
+     * @param <T>
+     */
+    private static class ActiveDescriptorStub<T> extends AbstractActiveDescriptor<T> {
+
+        private final T impl;
+
+        public ActiveDescriptorStub(T impl) {
+            this.impl = impl;
+        }
+
+
+        @Override
+        public Class<?> getImplementationClass() {
+            return impl.getClass();
+        }
+
+        @Override
+        public T create(ServiceHandle<?> serviceHandle) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            if (!super.equals(o)) return false;
+            ActiveDescriptorStub<?> that = (ActiveDescriptorStub<?>) o;
+            return Objects.equals(impl, that.impl);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(super.hashCode(), impl);
+        }
+    }
+
+    /**
+     * Stub implementation -- reason for this is to control the order of list of services in ServiceLocator.
+     *
+     * @param <T>
+     */
+    private static class ServiceHandleStub<T> implements ServiceHandle<T> {
```

Comment:
I guess that better than mocking the locator, you can just configure one for testing purposes, e.g. like here:
https://github.com/jersey/jersey/blob/master/core-common/src/test/java/org/glassfish/jersey/process/internal/ExecutorProvidersTest.java#L238-L256

To simulate user-registered exception mappers (if needed), you can create bindings using the "@Custom" qualifier:
https://github.com/jersey/jersey/blob/master/core-common/src/main/java/org/glassfish/jersey/internal/inject/Custom.java